examples/haproxy/haproxy.cfg

```diff
@@ -34,8 +34,10 @@ frontend test
     http-request return status 501 if { var(txn.berghain.error) -m found }
 
     acl berghain_valid var(txn.berghain.valid) -m bool
+    acl is_ssl ssl_fc
 
-    http-request return status 403 content-type "text/html" file "web/dist/index.html" if !berghain_valid !berghain_path berghain_active
+    http-request return status 403 content-type "text/html" file "web/dist/default/index.html" if !berghain_valid !berghain_path berghain_active !is_ssl
+    http-request return status 403 content-type "text/html" file "web/dist/native-crypto/index.html" if !berghain_valid !berghain_path berghain_active is_ssl
```

Comment:
Would it make sense to still use the non-native implementation with https for modern-enough browsers (probably using some user-agent ACLs, unless there is a cleaner way) or do you think both are equivalent speed-wise?

Reply:
Native should be always faster as it can do actual acceleration. I don't have benchmarks tho. I think simplicity wise its better to just keep it like this, because else you would need to have a compat list

Reply:
Or always have both in the frontend and switch to non-native if native isnt available

Reply:
ok, and one can always do such an ACL downstream if deemed better.

> Or always have both in the frontend and switch to non-native if native isnt available

how about something like the following?

```
-    if(import.meta.env.VITE_NATIVE_CRYPTO === "true"){
+    if(import.meta.env.VITE_NATIVE_CRYPTO === "true" || typeof BigInt === "undefined"){
```

Reply:
This wouldn't work as this is build time infrastructure. Vite renders two different htmls by replacing the env var and then stripping the code.

Reply:
I tested it and it returns the expected result in the output

default:

```
async function doHash(data) {
  if (typeof BigInt === "undefined") {
    return await nativeHash(data, "sha-256");
  } else {
    return bytesToHex(sha256(new TextEncoder().encode(data)));
  }
}
```

native-crypto:

```
async function doHash(data) {
  {
    return await nativeHash(data, "sha-256");
  }
}
```

Reply:
ah, if there is a better way to check for native crypto support, that would of course be neater.